### ISSUE TITLE
Build fails on hrev52025 (x86_64)

### ISSUE THREAD

**Issue by KevinAdams05** [HaikuArchives/BeMines#15]
Build fails on hrev52025 x86_64.
If I comment out the guts of BinaryInsertUnique (both methods) it will compile and appears to run fine. I don't know if that bit of code is even needed, or just extra since this appears to be a helper class from the BeOS days. The comment does say "DANGER, WILL ROBINSON!"  :)

Maybe there is a newer version of this class that has already been updated?

```
ObjectList.h: In member function 'bool BObjectList<T>::BinaryInsertUnique(T*, BObjectList<T>::CompareFunction)':
ObjectList.h:673:49: error: 'item' was not declared in this scope
  int32 index = _PointerList_::BinarySearchIndex(item,
                                                 ^~~~
ObjectList.h:673:49: note: suggested alternative: 'Items'
  int32 index = _PointerList_::BinarySearchIndex(item,
                                                 ^~~~
                                                 Items
ObjectList.h: In member function 'bool BObjectList<T>::BinaryInsertUnique(T*, BObjectList<T>::CompareFunctionWithState, void*)':
ObjectList.h:686:49: error: 'item' was not declared in this scope
  int32 index = _PointerList_::BinarySearchIndex(item,
                                                 ^~~~
ObjectList.h:686:49: note: suggested alternative: 'Items'
  int32 index = _PointerList_::BinarySearchIndex(item,
                                                 ^~~~
                                                 Items
/boot/system/develop/etc/makefile-engine:291: recipe for target 'objects.x86_64-cc7-release/App.o' failed

```

**Comment by KevinAdams05** [comment]
Fixed by Humdinger in bd1a954d8c0f0f0f8b60bca71965e06d360d19a8

**Comment by humdingerb** [comment]
Sorry, I didn't look for a ticket for this...
